openmetadata-ui/src/main/resources/ui/src/components/searched-data/SearchedData.interface.ts

```diff
@@ -63,14 +67,17 @@ export type SourceType = (
         | UserSearchSource
         | TeamSearchSource
         | TestCaseSearchSource
+        | StoredProcedureSearchSource
+        | DashboardDataModelSearchSource
       >,
       Fields
     >
 ) & {
-  id: string;
+  id?: string;
```

Comment:
Not sure why we are marking this as optional

Reply:
for the `StoredProcedure and DashboardDataModel` id is optional, that's why we have added this check.

Reply:
We should mark it required since other entity has required and avoid adding optional checks

Reply:
@chirag-madlani  changes made!